src/main/java/uk/gov/hmcts/reform/judicialbooking/domain/service/common/PrepareDataService.java

```diff
@@ -0,0 +1,79 @@
+package uk.gov.hmcts.reform.judicialbooking.domain.service.common;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.judicialbooking.data.BookingEntity;
+import uk.gov.hmcts.reform.judicialbooking.domain.model.Booking;
+import uk.gov.hmcts.reform.judicialbooking.domain.model.BookingResponse;
+import uk.gov.hmcts.reform.judicialbooking.domain.model.JudicialUserProfile;
+import uk.gov.hmcts.reform.judicialbooking.domain.model.OrmBooking;
+import uk.gov.hmcts.reform.judicialbooking.domain.model.OrmBookingAssignmentsRequest;
+import uk.gov.hmcts.reform.judicialbooking.domain.model.OrmBookingRequest;
+import uk.gov.hmcts.reform.judicialbooking.domain.model.enums.Status;
+
+import java.util.Collections;
+import java.util.List;
+
+@Service
+public class PrepareDataService {
+
+    public Booking prepareBooking(Booking booking, List<JudicialUserProfile> judicialUserProfiles) {
+
+        booking.setRoleId(judicialUserProfiles.get(0).getAppointments().get(0).getRoleId());
+        booking.setContractTypeId(judicialUserProfiles.get(0).getContractTypeId());
+        //TODO what to do when it is provided in initial request replace or compare?
+        booking.setBaseLocationId(judicialUserProfiles.get(0).getAppointments().get(0).getBaseLocationId());
+        //TODO booking.setRegionId(); this one requires another endpoint call I think
+        booking.setStatus(Status.NEW.toString());
+        return booking;
+    }
+
+    public OrmBookingAssignmentsRequest prepareOrmBookingRequest(Booking booking,
+                                                                 List<JudicialUserProfile> judicialUserProfiles) {
+        OrmBooking ormBooking = convertBookingToOrmBooking(booking);
+        OrmBookingRequest ormBookingRequest = OrmBookingRequest.builder()
+                .actorId(booking.getUserId())
+                .authorisations(judicialUserProfiles.get(0).getAuthorisations())
```

Comment:
There could be multiple authorisations for a single user, so we need to collect all the authorisation Ids in a list and capture here.